src/bin/pg_autoctl/monitor.c

```diff
@@ -2982,25 +3031,28 @@ monitor_stop_maintenance(Monitor *monitor, int nodeId)
 
 
 /*
- * monitor_get_notifications listens to notifications from the monitor.
+ * monitor_process_notifications listens to notifications from the monitor and
+ * calls a specific processing function for each notification received.
  *
- * Use the select(2) facility to check if something is ready to be read on the
- * PQconn socket for us. When it's the case, return the next notification
+ * We use the pselect(2) facility to check if something is ready to be read on
+ * the PQconn socket for us. When it's the case, return the next notification
  * message from the "state" channel. Other channel messages are sent to the log
  * directly.
  *
  * When the function returns true, it's safe for the caller to sleep, otherwise
  * it's expected that the caller keeps polling the results to drain the queue
  * of notifications received from the previous calls loop.
  */
-bool
-monitor_get_notifications(Monitor *monitor, int timeoutMs)
+static bool
+monitor_process_notifications(Monitor *monitor,
+							  int timeoutMs,
+							  char *channels[],
+							  void *NotificationContext,
```

Comment:
Looks like a type otherwise:
```suggestion
							  void *notificationContext,
```

Reply:
Good point, done now.